examples/Transfer_Learning/basic_tl.py

```diff
@@ -0,0 +1,176 @@
+## Transfer Learning
+
+# This example demonstrates BayBE's
+# {doc}`Transfer Learning </userguide/transfer_learning>` capabilities:
+# * We construct a campaign,
+# * give it access to data from a related but different task,
+# * and show how this additional information boosts performance.
+
+### Imports
+
+import json
+import os
+import sys
+from pathlib import Path
+from typing import Dict, List
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from botorch.test_functions.synthetic import Hartmann
+
+from baybe import Campaign
+from baybe.objective import Objective
+from baybe.parameters import NumericalDiscreteParameter, TaskParameter
+from baybe.searchspace import SearchSpace
+from baybe.simulation import simulate_scenarios
+from baybe.targets import NumericalTarget
+from baybe.utils.botorch_wrapper import botorch_function_wrapper
+
+### Settings
+
+# The following settings are used to set up the problem:
+
+SMOKE_TEST = "SMOKE_TEST" in os.environ
+DIMENSION = 3  # input dimensionality of the test function
+BATCH_SIZE = 1  # batch size of recommendations per DOE iteration
+N_MC_ITERATIONS = 5 if SMOKE_TEST else 75  # number of Monte Carlo runs
+N_DOE_ITERATIONS = 5 if SMOKE_TEST else 10  # number of DOE iterations
+POINTS_PER_DIM = 5 if SMOKE_TEST else 5  # number of grid points per input dimension
+
+### Defining the Tasks
+
+# For our example, we consider the "Hartmann" test function.
+#
+# More specifically, to demonstrate the transfer learning mechanism, we consider the
+# problem of optimizing the function using training data from its negated version.
+# The used model is of course not aware of this relationship but needs to infer it
+# from the data gathered during the optimization process.
+
+test_functions = {
+    "Test_Function": botorch_function_wrapper(Hartmann(dim=DIMENSION)),
+    "Training_Function": botorch_function_wrapper(Hartmann(dim=DIMENSION, negate=True)),
+}
+
+### Creating the Optimization Objective
+
+# Both test functions have a single output that is to be minimized.
+# The corresponding [Objective](baybe.objective.Objective)
+# is created as follows:
+
+objective = Objective(
+    mode="SINGLE", targets=[NumericalTarget(name="Target", mode="MIN")]
+)
+
+### Creating the Searchspace
+
+# The bounds of the search space are dictated by the test function:
+
+BOUNDS = Hartmann(dim=DIMENSION).bounds
+
+# First, we define one
+# [NumericalDiscreteParameter](baybe.parameters.numerical.NumericalDiscreteParameter)
+# per input dimension of the test function:
+
+discrete_params = [
+    NumericalDiscreteParameter(
+        name=f"x{d}",
+        values=np.linspace(lower, upper, POINTS_PER_DIM),
+    )
+    for d, (lower, upper) in enumerate(BOUNDS.T)
+]
+
+# ```{note}
```

Comment:
We once again need to discuss what should be part of examples vs. user guides :D In my opinion, such a note does not belong in an example but in a user guide. Also, we do not use *any* admonitions in other examples, so the style of this example is significantly different from our current examples.
Don't get me wrong - I am not saying that a different style is necessarily worse. We should only (once again) align on what we want to have in the examples vs. the user guides and whether having examples in different styles is fine.

Reply:
I think we've discussed this in the meeting, so feel free to resolve or discuss further with @Scienfitz

Reply:
Don't remember what we agreed upon in this specific case - do we keep the note? My memory tells me that we wanted to rather transform this, but my memory also isn't the best.

Reply:
So what I did after our discussion is to reduce the text in the note. I'll leave it now with Martin and you to decide what part of if to keep and what not. From my perspective, the note now contains the minimum information necessary for a user to understand the at first glance rather unintuitive choice WHY we optimize a continuous space using discrete parameters. Further, I feel that if we didn't mention the continuous alternative, this could give the wrong impression that baybe does not support continuous optimization. So I have the impression that if we cut it down even further, information will be lost. But I won't be here for the merge, so your choice 🙃

Reply:
i dont see anything against having the note, especially if its concise, lets just include and move on